### PR TITLE
OCPP update and clean up

### PR DESCRIPTION
**Pull request by matth-x**
General update of the OCPP lib and a few changes:
- Revision of the synchronization between OpenEVSE configs and OCPP configs (see #681)
- Lifecycle management of OCPP lib reworked and combined with `MicroTask.wake()` and `MicroTask.Infinate`
- Define OCPP factory defaults which reflect the typical usage of OpenEVSE better (addresses #663)
- Report the hardware maximum allowed current to the OCPP server (addresses #632)
- General clean up

### CODE REVIEW

**Review by chris1howell — DISMISSED** [review]

**Comment by jeremypoulter** [comment]
@chris1howell I am not sure this was ready to be merged. Was still in draft and has references to dev libraries

**Comment by jeremypoulter** [comment]
Oh sorry looks like it is not draft any more, will check.

**Comment by matth-x** [comment]
Thanks for taking a look. I left it as a draft because I was hoping for more real world tests before merging. Anyway, I will publish the OCPP lib to the PIO registry ASAP so you can merge.

File: src/ocpp.cpp [hunk 1]
```diff
@@ -42,99 +44,125 @@ void ArduinoOcppTask::begin(EvseManager &evse, LcdTask &lcd, EventLog &eventLog,
...
+    setOnReceiveRequest("ChangeConfiguration", [this] (JsonObject) {
+        DynamicJsonDocument updateQuery (JSON_OBJECT_SIZE(6));
+        updateQuery["ocpp_server"] = (const char*) *backendUrl;
+        updateQuery["ocpp_chargeBoxId"] = (const char*) *chargeBoxId;
+        updateQuery["ocpp_authkey"] = (const char*) *authKey;
+        updateQuery["ocpp_auth_auto"] = *freevendActive ? 1 : 0;
+        updateQuery["ocpp_idtag"] = (const char*) *freevendIdTag;
+        updateQuery["ocpp_auth_offline"] = *allowOfflineTxForUnknownId ? 1 : 0;
+        config_deserialize(updateQuery);
+        config_commit();
```
Comment (jeremypoulter) [inline src/ocpp.cpp:146]: Would be better to use `config_set` rather than using JSON

Comment (matth-x) [reply src/ocpp.cpp]: Done

**Comment by matth-x** [comment]
Now it should be ready to be merged

**Comment by matth-x** [comment]
@jeremypoulter Can you merge this? I tested it and it works as expected

**Comment by jeremypoulter** [comment]
Hi, I have been trying to get some time to review/test, made a start on it, but it is a big PR

**Comment by matth-x** [comment]
Okay, happy that it gets tested further. Do it when you find the time to, I'll keep the PR in sync.

**Review by jeremypoulter — APPROVED** [review]

**Pull request merged**